File: csharp/difference-of-squares/DifferenceOfSquares.cs
using System;

public static class DifferenceOfSquares {
    public static int CalculateSquareOfSum (int max) {
        int sum = ((max + 1) * (int) (max / 2)) + ((max % 2) == 0 ? 0 : (int) Math.Ceiling ((double) max / 2));
        return (int) Math.Pow (sum, 2);
    }

    public static int CalculateSumOfSquares (int max) {
        int sum = 0;
        for (int i = 1; i <= max; i++) {
            sum += (int) Math.Pow (i, 2);
        }
        return sum;
    }

    public static int CalculateDifferenceOfSquares (int max) {
        return CalculateSquareOfSum (max) - CalculateSumOfSquares (max);
    }
}

File: dart/difference-of-squares/lib/difference_of_squares.dart
import 'dart:math';

class DifferenceOfSquares {
  // Put your code here

  int squareOfSum(int num) => pow(
        List.generate(num, (int index) => index + 1)
            .fold(
              0,
              (int previousValue, int element) => previousValue + element,
            )
            .toDouble(),
        2,
      ).toInt();

  int sumOfSquares(int num) =>
      List.generate(num, (index) => pow(index + 1, 2).toInt())
          .fold(0, (previousValue, element) => previousValue + element);

  int differenceOfSquares(int num) => squareOfSum(num) - sumOfSquares(num);
}
